wrench/src/main.rs

```diff
@@ -417,116 +417,114 @@ fn main() {
             }
         }
 
-        let events = match window {
+        let event = match window {
             WindowWrapper::Headless(..) => {
-                vec![glutin::Event::Awakened]
+                glutin::Event::Awakened
             }
             WindowWrapper::Window(ref window, _) => {
-                window.poll_events().collect()
+                window.wait_events().next().unwrap()
             }
         };
 
-        for event in events {
-            if let Some(limit) = limit_seconds {
-                if (time::SteadyTime::now() - time_start) >= limit {
-                    break 'outer;
-                }
+        if let Some(limit) = limit_seconds {
+            if (time::SteadyTime::now() - time_start) >= limit {
+                break 'outer;
             }
+        }
 
-            match event {
-                glutin::Event::Awakened => {
-                    let (width, height) = window.get_inner_size_pixels();
-                    let dim = DeviceUintSize::new(width, height);
-                    wrench.update(dim);
+        match event {
+            glutin::Event::Awakened => {
+                let (width, height) = window.get_inner_size_pixels();
+                let dim = DeviceUintSize::new(width, height);
+                wrench.update(dim);
 
-                    let frame_num = thing.do_frame(&mut wrench);
-                    unsafe {
-                        CURRENT_FRAME_NUMBER = frame_num;
-                    }
+                let frame_num = thing.do_frame(&mut wrench);
+                unsafe {
+                    CURRENT_FRAME_NUMBER = frame_num;
+                }
 
-                    if show_help {
-                        wrench.show_onscreen_help();
-                    }
+                if show_help {
+                    wrench.show_onscreen_help();
+                }
+
+                wrench.render();
+                window.swap_buffers();
 
-                    wrench.render();
-                    window.swap_buffers();
+                let now = time::SteadyTime::now();
+                let dur = now - last;
 
-                    let now = time::SteadyTime::now();
-                    let dur = now - last;
+                min_time = min(min_time, dur);
+                min_min_time = min(min_min_time, dur);
```

Comment:
these (`min_min` and `max_max`) are not needed here since they are in the `if warmed_up` path below